javascript/MaterialXView/source/viewer.js

```diff
@@ -605,7 +605,7 @@ export class Material
         // and assign to the associatged geometry. If there are no looks
         // then the first material is found and assignment to all the
         // geometry. 
-        this._materials = [];
+        this._materials.length = 0;
```

Comment:
This was not being cleared properly when searching for new assignments so an old assignment (sucha as to "*") could be left behind as the last assignment.